tendermint/tests/support/lite/single_step_sequential/commit_tests.json

```diff
@@ -0,0 +1,2360 @@
+ 							"hash": "EC561779A1A1A2F969B11D41DF93878AA27AEACDEAAFC5A9564F52F8729FC91D",
+ 							"parts": {
+ 								"total": "1",
+ 								"hash": "198AB65E5090C2D65576D0F84876A97BDDCC8E50EBDB0EEA44D21B58CBC4660E"
+ 							}
+ 						},
+ 						"precommits": [
+ 							{
+ 								"type": 2,
+ 								"height": "1",
+ 								"round": "1",
+ 								"block_id": {
+ 									"hash": "EC561779A1A1A2F969B11D41DF93878AA27AEACDEAAFC5A9564F52F8729FC91D",
+ 									"parts": {
+ 										"total": "1",
+ 										"hash": "198AB65E5090C2D65576D0F84876A97BDDCC8E50EBDB0EEA44D21B58CBC4660E"
+ 									}
+ 								},
+ 								"timestamp": "2019-11-02T15:04:10Z",
+ 								"validator_address": "01F527D77D3FFCC4FCFF2DDC2952EEA5414F2A33",
+ 								"validator_index": "0",
+ 								"signature": "4TA6MKq3xvL+0BNy/AgETrRTq3TRidFyEEUe5lRAPysMAcBe3WSudbYHh6l+1Pu61foJLXiYz+wDYdrOrEk2Dw=="
+ 							}
+ 						]
+ 					}
+ 				},
+ 				"next_validator_set": {
+ 					"validators": [
+ 						{
+ 							"address": "01F527D77D3FFCC4FCFF2DDC2952EEA5414F2A33",
+ 							"pub_key": {
+ 								"type": "tendermint/PubKeyEd25519",
+ 								"value": "OAaNq3DX/15fGJP2MI6bujt1GRpvjwrqIevChirJsbc="
+ 							},
+ 							"voting_power": "50",
+ 							"proposer_priority": "0"
+ 						}
+ 					],
+ 					"proposer": {
+ 						"address": "01F527D77D3FFCC4FCFF2DDC2952EEA5414F2A33",
+ 						"pub_key": {
+ 							"type": "tendermint/PubKeyEd25519",
+ 							"value": "OAaNq3DX/15fGJP2MI6bujt1GRpvjwrqIevChirJsbc="
+ 						},
+ 						"voting_power": "50",
+ 						"proposer_priority": "0"
+ 					}
+ 				},
+ 				"trusting_period": "10800000000000",
+ 				"now": "2019-11-02T15:30:00Z"
+ 			},
+ 			"input": [
+ 				{
+ 					"signed_header": {
+ 						"header": {
+ 							"version": {
+ 								"block": "0",
+ 								"app": "0"
+ 							},
+ 							"chain_id": "test-chain-01",
+ 							"height": "2",
+ 							"time": "2019-11-02T15:04:10Z",
+ 							"num_txs": "0",
+ 							"total_txs": "0",
+ 							"last_block_id": {
+ 								"hash": "EC561779A1A1A2F969B11D41DF93878AA27AEACDEAAFC5A9564F52F8729FC91D",
+ 								"parts": {
+ 									"total": "1",
+ 									"hash": "198AB65E5090C2D65576D0F84876A97BDDCC8E50EBDB0EEA44D21B58CBC4660E"
+ 								}
+ 							},
+ 							"last_commit_hash": "AB6E7DD5B29A242104BAA90BE2D708A0A41C06139FB27D135B63905B3A5717D3",
+ 							"data_hash": "",
+ 							"validators_hash": "75B9F27F25F3515EF7CA61DD42C272CCC280C537E6A9ED493710EB361CB90B0D",
+ 							"next_validators_hash": "75B9F27F25F3515EF7CA61DD42C272CCC280C537E6A9ED493710EB361CB90B0D",
+ 							"consensus_hash": "048091BC7DDC283F77BFBF91D73C44DA58C3DF8A9CBC867405D8B7F3DAADA22F",
+ 							"app_hash": "6170705F68617368",
+ 							"last_results_hash": "",
+ 							"evidence_hash": "",
+ 							"proposer_address": "01F527D77D3FFCC4FCFF2DDC2952EEA5414F2A33"
+ 						},
+ 						"commit": {
+ 							"block_id": {
+ 								"hash": "2D2D2D2D5468697320697320612033322D6279746520737472696E672D2D2D2D",
```

Comment:
@Liamsi This is the "wrong header hash" in the commit which is passing this check 
` if header.hash() != commit.header_hash() {
        return Err(Error::InvalidCommitValue);
}`

Reply:
Awesome, thanks! Will look into it now.

Reply:
Can I see the code that generated this hash? `2D2D2D2D5468697320697320612033322D6279746520737472696E672D2D2D2D`

Reply:
Here it is: https://github.com/Shivani912/tendermint/blob/shivani-refactor/types/lite-client/generator/single_step_sequential_cases.go#L120

Reply:
That's really weird: I've added debug output to the pure `validate` function:
```rust
println!("header.hash()={:?}", header.hash());
println!("commit.header_hash()={:?}", commit.header_hash());
```
which yields:
```
header.hash()=Hash::Sha256(E3D4300A0C730F2912219F257314C4F3CE95E4E0954FB9F5E224B4FD076C51C0)
commit.header_hash()=Hash::Sha256(E3D4300A0C730F2912219F257314C4F3CE95E4E0954FB9F5E224B4FD076C51C0)
```

Reply:
Wait this test passes locally :-D It's `Case: one lite block, wrong PartsHeader.Total, with error` that is causing problems.

Reply:
This one is "description": "Case: one lite block, wrong header hash, with error"

Reply:
Ah, I got confused from the output of tests. Sorry!